### PR TITLE
[v1.14] Bump envoy to v1.27.x

### PR DESCRIPTION
**Pull request by sayboras**
* [ ] #27498 (@jrajahalme )
 * [x] #31221 (@sayboras)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 27498 31221
```

### CODE REVIEW

**Comment by sayboras** [comment]
/test-backport-1.14

**Comment by sayboras** [comment]
/test-backport-1.14

**Comment by sayboras** [comment]
/test-backport-1.14

**Comment by sayboras** [comment]
/test-backport-1.14

File: go.mod [hunk 1]
```diff
@@ -25,7 +25,7 @@ require (
 	github.com/cilium/fake v0.4.0
 	github.com/cilium/linters v0.0.0-20230711081823-012f25fa2197
 	github.com/cilium/lumberjack/v2 v2.3.0
-	github.com/cilium/proxy v0.0.0-20231202123106-38b645b854f3
+	github.com/cilium/proxy v0.0.0-20230911173141-49b6a7c9c075
```
Comment (aanm) [inline go.mod]: @sayboras this looks odd since we are downgrading the dependency.

**Review by aanm — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (sayboras) [reply go.mod]: Good point, seems like we have bump the version in previous backport https://github.com/cilium/cilium/pull/29447

**Comment by sayboras** [comment]
/test-backport-1.14

**Review by aanm — APPROVED** [review]

**Comment by sayboras** [comment]
/test-backport-1.13

**Comment by sayboras** [comment]
/test-backport-1.14

**Comment by sayboras** [comment]
/test-backport-1.14

**Review by squeed — APPROVED** [review]

**Pull request merged**